cypress/integration/post/Comment.feature

```diff
@@ -20,3 +20,25 @@ Feature: Post Comment
     Then my comment should be successfully created
     And I should see my comment
     And the editor should be cleared
+
+  Scenario: View short comments
+    Given I visit "post/bWBjpkTKZp/101-essays"
+    And I type in the following text:
+    """
+    This is a medium length comment--it is longer than 300 characters, which was the previous limit for a short message, but it will be less than 1200 characters, which is the new limit for a message that will not be abbreviated. This is just a little bit more text to push it past the limit of 300 characters.
+    """
+    And I click on the "Comment" button
+    Then my comment should be successfully created
+    And I should see the entirety of my comment
+    And the editor should be cleared
+
+  Scenario: View long comments
+    Given I visit "post/bWBjpkTKZp/101-essays"
+    And I type in the following text:
+    """
+    This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters.
+    """
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-05T11:16:50Z" title="Thursday, December 5th 2019, 12:16:50 pm +01:00">Dec 5, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -20,3 +20,25 @@ Feature: Post Comment
     Then my comment should be successfully created
     And I should see my comment
     And the editor should be cleared
+
+  Scenario: View short comments
+    Given I visit "post/bWBjpkTKZp/101-essays"
+    And I type in the following text:
+    """
+    This is a medium length comment--it is longer than 300 characters, which was the previous limit for a short message, but it will be less than 1200 characters, which is the new limit for a message that will not be abbreviated. This is just a little bit more text to push it past the limit of 300 characters.
+    """
+    And I click on the "Comment" button
+    Then my comment should be successfully created
+    And I should see the entirety of my comment
+    And the editor should be cleared
+
+  Scenario: View long comments
+    Given I visit "post/bWBjpkTKZp/101-essays"
+    And I type in the following text:
+    """
+    This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters. This is a very long comment, it is over 1200 characters.
+    """
```


this is really great @bocajnannerb , adding cypress tests is amazing!!
I have a suggestion though that could improve the code a bit... have you heard of `imperative` vs `declarative` with regards to testing/gherkin? Here is a short read about it https://wiki.saucelabs.com/display/DOCS/Best+Practice%3A+Imperative+v.+Declarative+Testing+Scenarios

I try to ask myself in these situations whether the information is necessary for a non-developer who might read this Scenario. Remember that our use of `cucumber` with our cypress means that we are creating executable documentation, where a non-developer/client could have a read of these scenarios and understand what they are meant to do.

So, one question I have is, is it important that the reader of this document know what was typed in the comment? or is it enough to know that it was a comment of longer than 300 characters, but shorter than 1200? or longer than 1200?
If the answer is that it's not important they know exactly what was typed, than we can extract this logic into our `steps` and "hide" it from people who are not developers, thus making it more `declarative`!!

So, you could do something similar to what @ogerly  did here https://github.com/Human-Connection/Human-Connection/blob/master/webapp/components/ContributionForm/ContributionForm.spec.js#L61.

What do you think?

Reply:
> [<img alt="jm-brennan" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/jm-brennan) **Authored by [jm-brennan](https://github.com/jm-brennan)**
_<time datetime="2019-12-06T05:52:06Z" title="Friday, December 6th 2019, 6:52:06 am +01:00">Dec 6, 2019</time>_

---


Well that explains why it was taking so long to run the integration stuff, I figured I must have done something wrong but didn't know what. I'll have to make a new pull request to do it from a branch, yes? And I will work to get the comment more hidden away, thank you for that recommendation